processing/src/main/java/io/druid/segment/StringDimensionMergerV9.java

```diff
@@ -167,18 +168,18 @@ public void writeMergedValueMetadata(List<IndexableAdapter> adapters) throws IOE
 
     cardinality = 0;
     if (numMergeIndex > 1) {
-      IndexMerger.DictionaryMergeIterator iterator = new IndexMerger.DictionaryMergeIterator(dimValueLookups, true);
+      dictionaryMergeIterator = new IndexMerger.DictionaryMergeIterator(dimValueLookups, true);
```

Comment:
Does this iterator really need to live beyond the execution of this one method? It doesn't seem to get used outside this method.